examples/tutorial_server_events.c

```diff
@@ -0,0 +1,199 @@
+    UA_Server_writeValue(server, bpr.targets[0].targetId.nodeId, value);
+
+    UA_BrowsePathResult_deleteMembers(&bpr);
+
+    /* message */
+    rpe.targetName = UA_QUALIFIEDNAME(0, "Message");
+    bpr = UA_Server_translateBrowsePathToNodeIds(server, &bp);
+    if (bpr.statusCode != UA_STATUSCODE_GOOD || bpr.targetsSize < 1) {
+        UA_LOG_WARNING(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND, "Event is missing message attribute.\n");
+        return bpr.statusCode;
+    }
+    UA_LocalizedText eventMessage = UA_LOCALIZEDTEXT("en-US", "An event has been generated.");
+    UA_Variant_setScalar(&value, &eventMessage, &UA_TYPES[UA_TYPES_LOCALIZEDTEXT]);
+    UA_Server_writeValue(server, bpr.targets[0].targetId.nodeId, value);
+
+    UA_BrowsePathResult_deleteMembers(&bpr);
+
+    return UA_STATUSCODE_GOOD;
+}
+
+/** Triggering an event
+ * ^^^^^^^^^^^^^^^^^^^^
+ * First a node representing an event is generated using ``setUpEvent``. Once our event is good to go, we specify
+ * a node which emits the event - in this case the server node. We can use ``triggerEvent`` to trigger our
+ * event onto said node. Note that once ``triggerEvent`` has been called, the node representation of the event is
+ * deleted.
+ */
+static UA_StatusCode
+generateEventMethodCallback(UA_Server *server,
+                         const UA_NodeId *sessionId, void *sessionHandle,
+                         const UA_NodeId *methodId, void *methodContext,
+                         const UA_NodeId *objectId, void *objectContext,
+                         size_t inputSize, const UA_Variant *input,
+                         size_t outputSize, UA_Variant *output) {
+    /* set up event */
+    UA_NodeId *eventNodeId = UA_NodeId_new();
+    if (!eventNodeId) {
+        UA_LOG_WARNING(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND,
+                       "Failed to allocate memory for eventId");
+        return UA_STATUSCODE_BADOUTOFMEMORY;
+    }
+
+    UA_StatusCode retval = setUpEvent(server, eventNodeId);
+    if (retval != UA_STATUSCODE_GOOD) {
+        UA_LOG_WARNING(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND,
+                       "Creating event failed. StatusCode %s", UA_StatusCode_name(retval));
+        return retval;
+    }
+
+    UA_NodeId serverId = UA_NODEID_NUMERIC(0, UA_NS0ID_SERVER);
+    UA_ByteString id;
+    retval = UA_Server_triggerEvent(server, eventNodeId, &serverId, &id);
+    if (retval != UA_STATUSCODE_GOOD) {
+        UA_ByteString_deleteMembers(&id);
+        return retval;
+    }
+
+    UA_Variant_setScalarCopy(output, &id, &UA_TYPES[UA_TYPES_BYTESTRING]);
+    UA_LOG_INFO(UA_Log_Stdout, UA_LOGCATEGORY_SERVER, "Generate Event was called.");
+    UA_NodeId_delete(eventNodeId);
+    UA_ByteString_deleteMembers(&id);
+    return retval;
+}
+
+/** Now, all that is left to do is to create a method node which uses our callback. We do not
+ * require any input and as output we will be using the `EventId` we receive from ``triggerEvent``.
```

Comment:
Add the following:
The `EventId` is generated by the server internally and is a random unique ID which identifies that specific event.